src/PopoverMenu.jsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import cx from 'classnames';
+
+/**
+ * @module PopoverMenu
+ */
+class PopoverMenu extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.selectedIndex = 0;
+		this.updateFocusBy = this.updateFocusBy.bind(this);
+		this.handleKeyUp = this.handleKeyUp.bind(this);
+	}
+
+	updateFocusBy(delta) {
+		const menuItems = findDOMNode(this).querySelectorAll('.popover-menu-item');
+		const targetIndex = this.selectedIndex + delta;
+
+		if (menuItems[targetIndex] == undefined) {
+			return;
+		}
+
+		this.selectedIndex = targetIndex;
+		menuItems[targetIndex].focus();
```

Comment:
this is a DOM-oriented approach to updating UI state, which sort of steps around the React component lifecycle and app structure conventions.

The main indication that something might not be quite right is that you're using `findDOMNode` in order to get an ordered list of particular `children`. It would probably be a better API to have `menuItems` be a `prop` on the `<PopoverMenu>` (propType: `React.PropTypes.arrayOf(React.PropTypes.instanceOf(PopoverMenuItem))` so that you don't have to select the corresponding nodes with `querySelectorAll`, and instead you can use `props.menuItems` directly.

Then, each menu item can have a `selected` prop that it can use to call `focus()` on its own DOM node - you should use the [`ref` prop](https://facebook.github.io/react/docs/refs-and-the-dom.html) on its rendered HTML elements to get the target element for `focus()`.

Advantages:

1. `PopoverMenu` doesn't need to know anything about the DOM
2. `PopoverMenuItem` is responsible for its own 'selected' UI state, which is prop-controllable
3. React component lifecycle stays intact because you'll need to use `setState({ selectedIndex: ... })` to get the menu items to re-render.
4. `PopoverMenuItem` won't be mixed in with other children (if that's even allowed? if it's not allowed to have other types of children, you can add a `propType` for `children` - it's currently ambiguous)